src/libraries/System.Private.Xml/src/System/Xml/Core/XsdValidatingReaderAsync.cs

```diff
@@ -506,13 +506,15 @@ private Task ProcessReaderEventAsync()
 
                 case XmlNodeType.Whitespace:
                 case XmlNodeType.SignificantWhitespace:
-                    _validator.ValidateWhitespace(GetStringValue);
-                    break;
+
+                    return GetValueAsync()
+                        .ContinueWith(t => _validator.ValidateWhitespace(t.GetAwaiter().GetResult()), TaskScheduler.Default);
 
                 case XmlNodeType.Text:          // text inside a node
                 case XmlNodeType.CDATA:         // <![CDATA[...]]>
-                    _validator.ValidateText(GetStringValue);
-                    break;
+
+                    return GetValueAsync()
+                        .ContinueWith(t => _validator.ValidateText(t.GetAwaiter().GetResult()), TaskScheduler.Default);
```

Comment:
These are going to incur costs associated with creating continuations even if the operation completes synchronously.  It'd be better to add a local static function like:
```C#
static async Task ValidateText(Task t) => _validator.ValidateText(await t.ConfigureAwait(false));
```
and then have the call site be:
```C#
return ValidateText(GetValueAsync());
```
Same for the ValidateWhitespace one above.

Reply:
Changed to use local static functions.